datalad_next/processors/lines_processor.py

```diff
@@ -0,0 +1,123 @@
+""" Generator the emits only complete lines """
+
+from __future__ import annotations
+
+from typing import (
+    Generator,
+    Iterable,
+)
+
+
+__all__ = ['lines_processor']
+
+
+def lines_processor(iterable: Iterable[bytes | str],
```

Comment:
Kept thinking about the name. Something active would be good. What about `itemize`?

Reply:
Not sure. It does assembling and splitting to yield only complete lines. Maybe `complete_lines`? But that seems a little too long. How about `lines`?

Reply:
A line somehow needs to come with a lineend. What about zero-bytes. Not a line, but an item.

Reply:
#538 has a variant with a more generic description. works well, I think.